cpp/src/io/orc/stripe_data.cu

```diff
@@ -866,6 +1019,15 @@ static __device__ uint32_t Integer_RLEv2(orc_bytestream_s* bs,
     __syncwarp();
   }
   __syncthreads();
+  if constexpr (cuda::std::is_same_v<T, int64_t>) {
```

Comment:
Is there a case where `T != int64_t` and `run_cache_manager_inst != nullptr`?

Reply:
There isn't supposed to be such a case with the current (dirty) fix. `run_cache_manager` is non-templated, and `write_to_cache`/`read_from_cache` expect `int64_t` data type. So I added the `if constexpr` there for the code to compile properly.

Reply:
In that case, it would be a good idea to add a comment here to avoid confusion

Reply:
Thanks! Completed.